sphinx/writers/latex.py

```diff
@@ -1360,25 +1376,10 @@ def depart_hlistcol(self, node):
         pass
 
     def latex_image_length(self, width_str):
-        """Convert `width_str` with rst length to LaTeX length.
-
-        This function is copied from docutils' latex writer
-        """
-        match = re.match('(\d*\.?\d*)\s*(\S*)', width_str)
-        if not match:
-            # fallback
-            return width_str
-        res = width_str
-        amount, unit = match.groups()[:2]
-        if not unit:
-            return None
-        elif unit == 'pt':
-            res = '%sbp' % amount  # convert to 'bp'
-        elif unit == "px":
-            res = "%.3f\\sphinxpxdimen" % (float(amount))
-        elif unit == "%":
-            res = "%.3f\\linewidth" % (float(amount) / 100.0)
-        return res
+        try:
+            return rstdim_to_latexdim(width_str)
+        except ValueError:
+            self.builder.warn('dimension unit %s is invalid. Ignored.')
```

Comment:
I am not good at Python, isn't a value for `%s` needed here ? (sorry if trivial question)
Besides, independently there is pre-existing check (I don't know from where), e.g.
```
/Users/foo/testallowpx/index.rst:13: ERROR: Error in "image" directive:
invalid option value: (option: "width"; value: '100.3.211')
not a positive measure of one of the following units:
"em" "ex" "px" "in" "cm" "mm" "pt" "pc" "%".

.. image:: sphinxwide.png
   :width: 100.3.211
```
This is issued only on first `make latex`. One must ``rm -fr _build/doctrees`` to regenerate error. But this is separate issue from this thread.